### ISSUE TITLE
PackageReference with no Version is not supported

### ISSUE THREAD

**Issue by KirillOsenkov** [mono/monodevelop#5892]
When loading a project with no Version in a PackageReference I hit an ArgumentNullException here:

![image](https://user-images.githubusercontent.com/679326/45247930-45d74180-b2c0-11e8-894b-9b924da88b96.png)

> VS bug [#679760](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/679760)

**Comment by Therzok** [comment]
There's a parameterless constructor for VersionRange: https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Versioning/VersionRange.cs#L60

**Comment by mrward** [comment]
Passing `*` seems correct to me but maybe it is not. Tried a simple restore test here just to compare an explicit `*` versus no version.

 - Test 1:

       <PackageReference Include="Newtonsoft.Json" Version="*" />

msbuild /t:restore - Here the latest Newtonsoft.Json 11.0.2 is used in the project.assets.json and there are no warnings.

 - Test 2:

       <PackageReference Include="Newtonsoft.Json" />

msbuild /t:restore - Here it seems that the oldest Newtonsoft.Json 3.5.8 is used. Restore has a warning NU1604. 

```
/Library/Frameworks/Mono.framework/Versions/5.12.0/lib/mono/msbuild/15.0/bin/NuGet.targets(114,5): warning : Project dependency Newtonsoft.Json does not contain an inclusive lower bound. Include a lower bound in the dependency version to ensure consistent restore results. 
```

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-13] Comment by **xamarinc**



Fixed in version 7.8.0.620 (master)



Pull Request #6090 merged by: Matt Ward  
 Author: mono  
 Commit: efb4c287c49b19bce7742c67a72dfb285b958592 (mono/monodevelop)

**Comment by mrward** [comment]
Fixed in version 7.7.0.1456 (release-7.7)

Pull request #6102
Commit: https://github.com/mono/monodevelop/commit/7aa2395211cfca209db0ee1c1b2af9576a83007c (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-22] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #6090 merged by: Matt Ward  
 Author: mono  
 Commit: efb4c287c49b19bce7742c67a72dfb285b958592 (xamarin/monodevelop)
